packages/opentelemetry-core/src/trace/DefaultSpan.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Span,
+  SpanContext,
+  Status,
+  Attributes,
+  SpanKind,
+} from '@opentelemetry/types';
+
+/** This is a no-op implementation of Span. */
+export class DefaultSpan implements Span {
+  name = 'DefaultSpan'; // default
+  spanId = '';
+  traceId = '';
+  parentSpanId = '';
+  kind = SpanKind.INTERNAL; // default
+  spanContext: SpanContext | null = null;
+
+  constructor() {
+    // TODO: https://github.com/open-telemetry/opentelemetry-js/pull/35
+    // this.spanId = randomSpanId();
+    // this.traceId = randomTraceId();
+  }
+
+  // By default returns a no-op SpanContext.
+  context(): SpanContext {
```

Comment:
Given that we allow `null` as a `spanContext` above, should this possibly return `null`? Alternatively, can we come up with some reasonable "default" span context where the values are set e.g. to empty strings or something? If this is dependent on the random span/trace ID, could you add a TODO about that?

Reply:
The idea is actually to not accept `null` to avoid every caller needing to add a condition. So as you mentioned, I would go with a default `SpanContext`.

Reply:
We should not accept null here.